packages/access/script/generateVoucherCodes.js

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env node
+/**
+ * This script can generated a desired amount of access grant voucher
+ * codes for a single user.
+ *
+ * Usage:
+ * packages/access/script/generateVoucherCodes.js --campaign <A Campaign ID> --granter <A User ID> --prefix ABC --amount 10
+ */
+
+require('@orbiting/backend-modules-env').config()
+const yargs = require('yargs')
+
+const PgDb = require('@orbiting/backend-modules-base/lib/pgdb')
+
+const grantsLib = require('../lib/grants')
+
+const argv = yargs
+  .option('campaign', { alias: ['c', 'ac', 'accessCampaignId'] })
+  .option('granter', { alias: ['g', 'u', 'user'] })
+  .option('amount', {
+    alias: ['n'],
+    type: 'number',
+    default: 1
+  })
+  .option('prefix', { alias: ['p'] })
+  .default(['campaign', 'granter', 'label'])
+  .check(argv => {
+    if (argv.prefix && argv.prefix.length >= grantsLib.VOUCHER_CODE_LENGTH) {
+      return `Check --prefix. "${argv.prefix}" too long to generate voucher codes.`
+    }
+
+    if (argv.prefix.match(/[^123456789ABCDEFGHKLMRSTUWXYZ]+/)) {
+      return `Check --prefix. "${argv.prefix}" contains invalid chars.`
+    }
+
+    if (argv.amount < 1) {
+      return `Check --amount. Should be at least 1.`
+    }
+
+    return true
+  })
+  .help()
+  .version()
+  .argv
+
+PgDb.connect().then(async pgdb => {
+  const transaction = await pgdb.transactionBegin()
+
+  try {
+    const granter = await pgdb.public.users.findOne({ id: argv.granter })
+
+    if (!granter) {
+      throw new Error('User not found.')
+    }
+
+    const grants = []
+
+    let iteration = 0
+    while (grants.length < argv.amount && iteration < 5 * argv.amount) {
+      iteration++
+
+      const grant = {
+        voucherCode: await pgdb.queryOneField(`SELECT make_hrid('"accessGrants"'::regclass, 'voucherCode'::text, ${grantsLib.VOUCHER_CODE_LENGTH}::bigint)`)
```

Comment:
why not use the transaction here? Then you could skip the check if it's already in the table on L71.